docs/DeveloperGuide.md

```diff
@@ -298,6 +298,87 @@ The following sequence diagram shows how deleting a session works:
+compatible recurring session otherwise, is handled through `AddRecurringSessionParser#parse(input)`, with key checks
+depicted below.
+![AddRecurringSessionCommandParserActivityDiagram](images/shion/AddRecurringSessionCommandParserActivityDiagram.png)
+
+##### Design Considerations
+Aspect 1: Use of Inheritance for RecurringSession and related classes
+* **Alternative 1 (current choice)**: Both `RecurringSession` and `RecurringSessionCommand` classes extends 
+  `Session` and `SessionCommand` classes
+    * Pros:
+        * Promotes "don't repeat yourself" principle, given the that recurring session is a proper subtype of superclass session.
+          Enhancements to recurring session is only additive to superclass Session.
+        * Reduced complexity. Abstract classes/interfaces are not required with the simple inheritance.
+    * Cons:
+        * Uses subclass coupling; coupling is increased. Reduces expansion and maintainability, in the possible event
+    of redefining recurring session and session relationship.
+
+* **Alternative 2**: Creating a Session interface to be implemented by Single-Session class and Recurring Session.
+    * Pros:
+        * Eliminates subclass coupling.
+    * Cons:
+        * Additional lines of code, at current implementation where Recurring Session shares the same logical domain as
+    Session and is a proper subtype of Session.
+          
+Alternative 1 was chosen for the practical consideration that less code was necessary for the forseeable requirements
+given the current set of use cases. Though increasing coupling, the inheritance relationship is simple and sufficient,
+as methods to `RecurringSession` is strictly only additive to those is `Session`. The increased coupling does not
+complicated testing as well, due to the fact that every test on `Session` applies to `RecurringSession`.
+
+
+Aspect 2: Brute or mathematical implementation for overlap checks involving recurring sessions
+* **Alternative 1 (current choice)**: Brute-force approach to checking for overlap with recurring sessions.
+    * Pros:
+        * Logically simple, hence easy to implement, test, review.
+    * Cons:
+        * Worst case scenario has no bounds.
+
+* **Alternative 2**: Mathematical approach to checking for overlap with recurring session, using ideas of GCD
+  and extended euclid algorithm
+    * Pros:
+        * Fixed upper bound to amount of computation; better (bounded) worst case.
+    * Cons:
+        * Not straightforward, may require domain-specific knowledge, i.e. Math
+    
+Alternative 1 was chosen as a worst case scenario that harms is effectively non-existent.
+Realistic, practical user inputs will involve recurrence intervals that can be perceived by the human brain of the
+user; this will not cause computations that give significant run-time, even when approach is burte.
+Alternative 2 would require formal verification, which can be time-consuming.
+
```

Comment:
Nice addition here! But I think you should mention a bit about where this is done. Like in the hasOverlappingSession() method, so that the reader can follow more easily.

Reply:
ok will do thanks!